### PR TITLE
add maptalks typings

### PR DESCRIPTION
**Pull request by yanyu510**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/maptalks/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5191 @@
+// Type definitions for maptalks 0.49
+// Project: https://github.com/maptalks/maptalks.js
+// Definitions by: Yu Yan <https://github.com/yanyu510>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
```
Comment (peterblazejewicz) [inline types/maptalks/index.d.ts:5]:
```suggestion
```
you can remove that safely, no longer used in DT ecosystem

**Comment by peterblazejewicz** [comment]
You may want torun prettier on those, that would remove some exclusions you've added to `tslint` configuration:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes

**Comment by yanyu510** [comment]
> You may want torun prettier on those, that would remove some exclusions you've added to `tslint` configuration:
> https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes

Thanks, I removed all rules from `tslint` configuration

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@yanyu510 thx!

**Review by peterblazejewicz — APPROVED** [review]

**Pull request merged**